### PR TITLE
Add more fail statement for error cases

### PR DESCRIPTION
**Pull request by seratch**
A portion of the tests for error cases doesn't have `fail` statements when an excepted exception is not thrown. This pull request improves those tests.

### CODE REVIEW

**Comment by seratch** [comment]
```
[ERROR] /home/travis/build/salesforce/mt-dynamo/src/test/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMappingTest.java:16: Using the '.*' form of import should be avoided - org.junit.jupiter.api.Assertions.*. [AvoidStarImport]
```

OK, will fix it.

File: src/test/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMappingTest.java [hunk 1]
```diff
@@ -360,7 +360,7 @@ void validatePhysicalTable() {
     private static void assertException(TestFunction test, String expectedMessagePrefix) {
         try {
             test.run();
-            throw new RuntimeException("expected exception '" + expectedMessagePrefix + "' not encountered");
+            fail("It was expected exception '" + expectedMessagePrefix + "' not encountered");
```
Comment (horwitzj) [inline src/test/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMappingTest.java]: pleases replace "It was expected" with "Expected"

File: src/test/java/com/salesforce/dynamodbv2/mt/mappers/index/DynamoSecondaryIndexMapperByTypeImplTest.java [hunk 2]
```diff
@@ -56,6 +57,7 @@ void testNoMatch() {
             GSI)));
         try {
             new DynamoSecondaryIndexMapperByTypeImpl().lookupPhysicalSecondaryIndex(vsi, physicalTable);
+            fail("It was expected that #lookupPhysicalSecondaryIndex throws MappingException");
```
Comment (horwitzj) [inline src/test/java/com/salesforce/dynamodbv2/mt/mappers/index/DynamoSecondaryIndexMapperByTypeImplTest.java]: instead, please use "Expected MappingException not thrown" (won't "lookupPhysicalSecondaryIndex" already appear in the stack trace?)—similarly for other changes

**Comment by horwitzj** [comment]
some great catches—thanks! (I'll finish reviewing later today.)

**Comment by seratch** [comment]
@horwitzj Thank you for your review. I've updated the changes.

**Pull request merged**